include/seqcmd.h

```diff
@@ -0,0 +1,510 @@
+    Audio_QueueSeqCmd((SEQCMD_OP_QUEUE_SEQUENCE << 28) | ((u8)(seqPlayerIndex) << 24) | ((u8)(fadeInDuration) << 16) | \
+                      ((u8)(priority) << 8) | (u8)(seqId))
+
+/**
+ * Remove a sequence from a queue of sequences associated with a given seqPlayer.
+ * If the sequence is first in queue, stop the sequence
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to queue the sequence
+ * @param fadeInDuration if the sequence is stopped, gradually fade out over a duration, gradually fade in next seq
+ * @param seqId the id of the sequence to remove, see NA_BGM_XXX
+ *
+ * @note for the next sequence in the queue to play, the current sequence must be unqueued with this command
+ */
+#define SEQCMD_UNQUEUE_SEQUENCE(seqPlayerIndex, fadeOutInDuration, seqId)                 \
+    Audio_QueueSeqCmd((SEQCMD_OP_UNQUEUE_SEQUENCE << 28) | ((u8)(seqPlayerIndex) << 24) | \
+                      ((u8)(fadeOutInDuration) << 16) | (seqId))
+
+/**
+ * Set the volume of an entire sequence on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param duration the duration to set the volume over
+ * @param volume the target volume for the sequence
+ */
+#define SEQCMD_SET_PLAYER_VOLUME(seqPlayerIndex, duration, volume)                                              \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_PLAYER_VOLUME << 28) | ((u8)(seqPlayerIndex) << 24) | ((duration) << 16) | \
+                      (volume))
+
+/**
+ * Scale the frequency of an entire sequence on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param duration the duration to set the frequency over
+ * @param freq the scaling factor to shift the pitch, relative to 1000.
+ *
+ * @note 2000 will double the frequency (raise an octave), 500 will half the frequency (lower an octave)
+ */
+#define SEQCMD_SET_PLAYER_FREQ(seqPlayerIndex, duration, freq) \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_PLAYER_FREQ << 28) | ((u8)(seqPlayerIndex) << 24) | ((duration) << 16) | (freq))
+
+/**
+ * Scale the frequency of a specific channel on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelIndex the index of the channel to modify
+ * @param duration the duration to set the frequency over
+ * @param freq the scaling factor to shift the pitch, relative to 1000.
+ *
+ * @note a frequency of 2000 will double the frequency (raise an octave), 500 will half the frequency (lower an octave)
+ */
+#define SEQCMD_SET_CHANNEL_FREQ(seqPlayerIndex, channelIndex, duration, freq)                                  \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_CHANNEL_FREQ << 28) | ((u8)(seqPlayerIndex) << 24) | ((duration) << 16) | \
+                      ((channelIndex) << 12) | (freq))
+
+/**
+ * Set the volume of a specific channel on a given seqPlayer over a specified duration
```

Comment:
This reads like the volume will be "set" for "a specified duration", instead of """fade to the requested volume over the specified duration"""